client/app/scripts/superdesk-publish/filters/publish-filters.js

```diff
@@ -580,21 +576,29 @@ function FilterSearchController($scope, filters, notify) {
     }
 
     $scope.search = function() {
-        $scope.searchResult = null;
-        $scope.filterCondition.value = getFilterValue();
-        var inputs = {
-            'field': $scope.filterCondition.field,
-            'operator': $scope.filterCondition.operator,
-            'value': $scope.filterCondition.value
-        };
-
-        filters.getFilterSearchResults(inputs).then(function(result) {
-            $scope.searchResult = result;
-            if (result.length === 0) {
-                notify.error(gettext('no results found'));
-            }
-        });
+        if (!$scope.loading) {
+            $scope.searchResult = null;
+            $scope.filterCondition.value = getFilterValue();
+            var inputs = {
+                'field': $scope.filterCondition.field,
+                'operator': $scope.filterCondition.operator,
+                'value': $scope.filterCondition.value
+            };
+            $scope.loading = true;
+            filters.getFilterSearchResults(inputs).then(function(result) {
+                $scope.searchResult = result;
+                if (result.length === 0) {
+                    notify.error(gettext('no results found'));
+                }
+                $scope.filterCondition.value = null;
+                $scope.loading = false;
```

Comment:
I think it is good practice to have this line "scope.loading = false" in a finally block so if will invalidate even if there's an error.